mmengine/registry/registry.py

```diff
@@ -431,8 +431,17 @@ def get(self, key: str) -> Optional[Type]:
             >>> mobilenet_cls = DETECTORS.get('cls.MobileNet')
         """
         # Avoid circular import
+        if not isinstance(key, str):
+            raise TypeError(
+                'The key argument of `Registry.get` must be a str, '
+                f'got {type(key)}')
         from ..logging import print_log
-
+        try:
+            obj_cls = locate(key)
+        except Exception:
+            raise RuntimeError(f'Failed to get {key}')
+        if obj_cls is not None:
+            return obj_cls
```

Comment:
Please update the docstring of `get` method.

Reply:
Actually, it's strange to implement this code snipte here, and let me do some explanation. If we want to build the model using a configuration like `dict(type='mmengine.model.BaseModel')`, which can be dumped by lazy import config, we need the this code snippet for `Registry.get` to work. In this scenario, the instance obtained from `Registry.get` can be any object, not necessarily a strict class or callable object.

In your case, if `mmengine.model.BaseModel.training` is passed in, a `None` object will be return since `training` is not a class attribute for `BaseModel`

I'll add some docstring and comment here.